pymc3/distributions/continuous.py

```diff
@@ -162,8 +162,16 @@ def transform_params(rv_var):
             if cls.bound_args_indices[1] is not None:
                 upper = args[cls.bound_args_indices[1]]
 
-            lower = at.as_tensor_variable(lower) if lower is not None else None
-            upper = at.as_tensor_variable(upper) if upper is not None else None
+            lower = (
+                None
+                if lower is None or (isinstance(lower, TensorConstant) and lower.value == -np.inf)
```

Comment:
I think this will raise an error if you get a non scalar constant in other contexts

Reply:
Good point! I added a test for that, which is currently failing `pymc3.tests.test_distributions.TestBoundedContinuous.test_missing_upper_bound_array` still need to fix it though, I should be able to do it later today. I guess it will be a matter of having a check for scalar and one for arrays. btw I should also create an array of np.infs within `dist` when the other params are arrays.

Reply:
By the way, I have been working on something related for Censored distributions, and this is how I wrote the "constant inf check" so far: https://github.com/aesara-devs/aeppl/blob/46fac1347d129e0a6c7391c87bbae456f51aa009/aeppl/truncation.py#L36

I am not very happy with the solution. I am still thinking whether it makes more sense to subclass (in my case the Censored class), in your case the Truncated Normal, into left-truncated, right-truncated, and truncated, depending on the bounds.

Reply:
Thanks for sharing the link, I will give it a look later.

Possibly it may just be as simple as doing as per below:
```python
lower = (
                at.as_tensor_variable(lower)
                if lower is not None
                and (isinstance(lower, TensorConstant) and -np.inf not in lower.value)
                else None
            )
```

I pushed an updated (and not too ugly) approach that is passing the test with arrays. I need to check it more thoroughly though

Reply:
Can't read that code in my mind right now, but I am thinking of the case where you have lower = [-np.inf, 0]: when one of the replications of the variable is lower bounded and the other is not, in which case the logp and transform should treat the variable as being lowerbounded.

Reply:
Good point, I only realized it after sharing the snippet. The approach with `np.all` is pretty neat! thanks for sharing it